test/GroupByTest.cpp

```diff
@@ -1912,3 +1911,282 @@ TEST(GroupBy, Descriptor) {
+                              makeIdTableFromVector({{1, 6}}, IntId),
+                              makeIdTableFromVector({{2, 24}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyImplicitGroupByWorksWithEmptyInput) {
+  std::vector<IdTable> idTables;
+  idTables.emplace_back(1, qec_->getAllocator());
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x"), true, std::vector<ColumnIndex>{0});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<1>(groupBy, {makeIdTableFromVector({{0}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyGroupsOfSize1AreAggregatedCorrectly) {
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1, 0}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{2, 1}, {4, 2}, {8, 3}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{16, 4}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{32, 5}, {64, 6}}, IntId));
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x", "?y"), true,
+      std::vector<ColumnIndex>{1});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {V{"?y"}}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<4>(
+      groupBy, {makeIdTableFromVector({{0, 1}, {1, 2}, {2, 4}}, IntId),
+                makeIdTableFromVector({{3, 8}}, IntId),
+                makeIdTableFromVector({{4, 16}, {5, 32}}, IntId),
+                makeIdTableFromVector({{6, 64}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyNestedAggregateFunctionsWorkCorrectly) {
+  using L = TripleComponent::Literal;
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1, 0}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{2, 1}, {4, 1}, {8, 2}}, IntId));
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x", "?y"), true,
+      std::vector<ColumnIndex>{1});
+
+  std::vector<std::unique_ptr<SparqlExpression>> children;
+  children.push_back(makeSum("?x"));
+  children.push_back(std::make_unique<StringLiteralExpression>(
+      L::fromStringRepresentation("\"---\"")));
+  Alias alias{SparqlExpressionPimpl{makeConcatExpression(std::move(children)),
+                                    "CONCAT(SUM(?x), \"---\")"},
+              V{"?result"}};
+  GroupBy groupBy{qec_, {V{"?y"}}, {std::move(alias)}, std::move(subtree)};
+  auto result = groupBy.computeResultOnlyForTesting(GetParam());
+
+  auto makeEntry = [&result](std::string string) {
```

Comment:
As this test has a unique and complicated setup (which is required here), 
Add a comment that describes, what the expression is here (similar to the above, where the input was `x, y` and you did `Sum(?x) GROUP BY ?y`, what is the corresponding expression here? (also describe the inputs in human-readable language, the initialization of Literals is always a bit complicated to read.